### ISSUE TITLE
VS 2017.4 Preview deadlocks on Test 10964.

### ISSUE THREAD

**Issue by garyfurnish** [boostorg/thread#132]
VS 2017.4 Preview (and I believe 2017.3) deadlock mysteriously running test 10964.  Interestingly enough, this does not create a failure in the tests, but does prevent advancement of b2 until task manager is used to end-task. This was tested in 3222938.  See [test_10964.cpp](https://github.com/boostorg/thread/blob/master/test/test_10964.cpp)

**Comment by viboes** [comment]
Sorry, could you tell me if VS 2017.4 Preview (or 2017.3) are run in http://www.boost.org/development/tests/develop/developer/thread.html?

**Comment by garyfurnish** [comment]
This was run on
```
OS Name	Microsoft Windows 10 Home Insider Preview
Version	10.0.16275 Build 16275
Processor	Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz, 3401 Mhz, 4 Core(s), 8 Logical Processor(s)
Installed Physical Memory (RAM)	16.0 GB
```
The closest in the bench is http://www.boost.org/development/tests/develop/teeks99-09-p-win2012R2-64on64.html as far as I can tell  (boost considers all post 14 as 14.1 so this is not helpful.  This It could be as early as 2017 (no update))  2017.4 Preview I suspect is not.  Its also only run on windows 2012 in the test bench, so there could be some significant divergence in behavior potentially.  Also Update 3 changed it to default to c++latest, so that could also introduce divergence.

**Comment by garyfurnish** [comment]
These bug also seem to persist across rebuilds.

**Comment by viboes** [comment]
Do you have an idea of the VS differences between the versions that work and the ones that don't work?

What do you mean by defaults to c++latest? C++17?

**Comment by garyfurnish** [comment]
Yes.  It defaults to C++17 + MS stuff.  The differences are not obvious beyond "improved conformance" and more ts stuff.  One of the big ones is autoptr was officially removed, but I didn't see any breakage from that.  Update 3 notes are: https://www.visualstudio.com/en-us/news/releasenotes/vs2015-update3-vs .

**Comment by garyfurnish** [comment]
Tests clean and closing as fixed by 39c23dd.
